services/w3c/w3c-validation.tester.js

```diff
@@ -0,0 +1,87 @@
+'use strict'
+
+const t = (module.exports = require('../tester').createServiceTester())
+
```

Comment:
In order to minimize brittleness in our service tests, we don't typically assert an _exact_ value for the message. The common approach is to use `@hapi/joi` to assert that the message is what we expected. 

Here's a few examples from other services:
https://github.com/badges/shields/blob/master/services/sonar/sonar-violations.tester.js#L6-L11
https://github.com/badges/shields/blob/master/services/github/github-hacktoberfest.tester.js
https://github.com/badges/shields/blob/ce0ddf93fc75e905743ac40db55c62a9bf44a53c/services/symfony/symfony-insight-violations.tester.js

Reply:
Done I made the messages use Joi.string().regex checks

Reply:
Excellent. Could we take this another step and do something like:

```js
const isW3CMessage = Joi.alternatives().try('validated', isErrorOnly, isWarningOnly, isErrorAndWarning),
```

I'm not familiar with the targetUrl's that are being used in the tests, nor if/when the results of the W3C checker may change for this service. If you feel pretty confident that the results for those targets will be static for a long period of time, then it would be good to include some inline comments/links to details about those targets, etc. for future readers.

If the results for those pages are a bit more dynamic (may change at any time/will likely change within the next several months/etc.), then it might be better to remove the `color` checks from these tests, and change the message to be `message: isW3CMessage,` and then create a .spec file and add some unit tests for the `render` function to validate the color

Reply:
The pages being tested again are supposed to be static. But how about making the colors `const isW3CColor = Joi.alternatives().try('brightgreen','red','yellow')`

Reply:
Sure, though I do think it would be good to have some tests that validate the color and messaging logic

Reply:
I made the changes to the tests

Reply:
There are unit tests in w3c-validation-helper.spec.js under the `getColor` and `getMessage` test suites to cover the combinations of MessageTypes derived from the response of the API for both the colors and messages.

Example

```
    it('returns "2 errors, 2 warnings" if 2 error and 2 warning message is provided', function() {
      const messageTypes = { error: 2, warning: 2 }

      const actualResult = getMessage(messageTypes)

      expect(actualResult).to.equal('2 errors, 2 warnings')
    })
```

Reply:
> There are unit tests in w3c-validation-helper.spec.js under the getColor and getMessage test suites to cover the combinations of MessageTypes derived from the response of the API for both the colors and messages.

Gotcha, don't worry about the render function then

> I am thinking of pulling out the logic from the transform method into a helper function and unit test that. What do you think?

it's fine to leave that inline in the same file as the service class definition. it's helpful to have the schema defining what the response object will look like in the same place as the code that's dealing with that response